packages/scm/src/browser/decorations/scm-decorations-service.ts

```diff
@@ -69,7 +69,8 @@ export class ScmDecorationsService {
         const currentRepo = this.scmService.selectedRepository;
         if (currentRepo) {
             try {
-                const uri = editor.uri.withScheme(currentRepo.provider.id).withQuery(`{"path":"${editor.uri['codeUri'].fsPath}","ref":"~"}`);
+                const query = { path: editor.uri['codeUri'].fsPath, ref: '~' };
+                const uri = editor.uri.withScheme(currentRepo.provider.id).withQuery(JSON.stringify(query));
```

Comment:
I'll need an explanation of why this is the correct incantation. What does this do? It's obscure enough to at lease warrant a comment.

Reply:
It ensures that backslashes in fsPath are escaped in the query JSON, as the commit message says.

The previous version tried to construct the JSON string manually, and failed to account for the necessary escaping. Now `JSON.stringify` is used to do it properly.